ocs_ci/framework/pytest_customization/ocscilib.py

```diff
@@ -662,6 +662,7 @@ def pytest_collection_modifyitems(session, config, items):
 
 @pytest.hookimpl(tryfirst=True, hookwrapper=True)
 def pytest_runtest_makereport(item, call):
+    ocsci_config.RUN["number_of_tests"] = len(item.session.items)
```

Comment:
pytest_runtest_makereport is called after every item is executed IIUC.
```
Called to create a TestReport for each of the setup, call and teardown runtest phases of a test item.
```
[Documentation link](https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.TestReport)

So for every test we run again and again right? I thought this should be one time operation done in different hook called:
pytest_collection_modifyitems

Where at the end you can do len on the items I think only once at the end of this hook.

But I might see a problem that we have this hook implementation in many different places:
```
tests/conftest.py:def pytest_collection_modifyitems(session, items):
tests/disaster-recovery/mdr/conftest.py:def pytest_collection_modifyitems(items):
tests/disaster-recovery/regional-dr/conftest.py:def pytest_collection_modifyitems(items):
tests/disaster-recovery/sc_arbiter/conftest.py:def pytest_collection_modifyitems(items):
tests/e2e/conftest.py:def pytest_collection_modifyitems(items):
tests/e2e/scale/conftest.py:def pytest_collection_modifyitems(items):
tests/manage/mcg/conftest.py:def pytest_collection_modifyitems(items):
tests/manage/pv_services/conftest.py:def pytest_collection_modifyitems(items):
tests/manage/rgw/conftest.py:def pytest_collection_modifyitems(items):
tests/manage/storageclass/conftest.py:def pytest_collection_modifyitems(items):
```

So maybe this is the reason why you haven't use it.

What about this hook:
https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.hookspec.pytest_collection_finish

? This might be after all collection is done.

Reply:
Will use pytest_collection_finish